### PR TITLE
Add more MLX dispatches

### PR DESCRIPTION
**Pull request by jessegrabowski**
## Description
PR adds MLX dispatches for:

- CumOp
- Repeat
- SortOp
- ArgSortOp
- LogSoftmax

I went through the list of Ops in `jax.linker.dispatch.extra_ops.py`, looking for more stuff to add, but didn't find much in the `mlx.core` namespace. I checked for `searchsorted`, `unique`, `bartlett`, and `diff`, but didn't find anything. Maybe I looked in the wrong place?

## Related Issue

- [ ] Closes #
- [ ] Related to #1350 

## Checklist

- [ ] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [x] Included tests that prove the fix is effective or that the new feature works
- [ ] Added necessary documentation (docstrings and/or example notebooks)
- [ ] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Type of change

- [x] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

----
📚 Documentation preview 📚: https://pytensor--1684.org.readthedocs.build/en/1684/

### CODE REVIEW

File: pytensor/link/mlx/dispatch/sort.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+from pytensor.link.mlx.dispatch.basic import mlx_funcify
+from pytensor.tensor.sort import ArgSortOp, SortOp
+
+
+@mlx_funcify.register(SortOp)
+def mlx_funcify_Sort(op, **kwargs):
+    kind = op.kind
+
+    def sort(x, axis, kind=kind):
+        if kind != "quicksort":
```
Comment (ricardoV94) [inline pytensor/link/mlx/dispatch/sort.py]: In other backends we just warn it will be ignored, but doesn't matter much, I don't think people customize it much

**Review by ricardoV94 — APPROVED** [review]

File: pytensor/link/mlx/dispatch/extra_ops.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    return cumop
+
+
+@mlx_funcify.register(Repeat)
+def jax_funcify_Repeat(op, **kwargs):
+    axis = op.axis
+
+    def repeat(x, repeats, axis=axis):
+        if not isinstance(repeats, int):
```
Comment (ricardoV94) [inline pytensor/link/mlx/dispatch/extra_ops.py:29]: This is known at dispatch time, raise then?

File: pytensor/link/mlx/dispatch/sort.py (hunk 1 above)
Comment (ricardoV94) [inline pytensor/link/mlx/dispatch/sort.py]: Also known at dispatch time

File: pytensor/link/mlx/dispatch/extra_ops.py (hunk 2 above)
Comment (jessegrabowski) [reply pytensor/link/mlx/dispatch/extra_ops.py:29]: `repeats` is a symbolic input. We only know axis as dispatch time.

Comment (jessegrabowski) [reply pytensor/link/mlx/dispatch/extra_ops.py:29]: It's a weird mlx-specific limitation. There might be a work-around, but I don't want to do it in this PR. Just getting some common cases covered.

Comment (ricardoV94) [reply pytensor/link/mlx/dispatch/extra_ops.py:29]: you know whether repeats is 0d or 1d at dispatch time? actually isn't the op always 1d now and we use alloc for 0d?

Comment (jessegrabowski) [reply pytensor/link/mlx/dispatch/extra_ops.py:29]: How? It's a symbolic input

Comment (jessegrabowski) [reply pytensor/link/mlx/dispatch/extra_ops.py:29]: Oh by checking `op.inputs[1]` :fivehead:

Comment (ricardoV94) [reply pytensor/link/mlx/dispatch/extra_ops.py:29]: by checking node.inputs. Node is the second argument to all dispatches
